### PR TITLE
#74,75 Creating and using bucket-fs connection objects

### LINKED ISSUE #74: Add a CLI callback function for creating a connection object with BFS credentials

### LINKED ISSUE #75: Add a function to create a bucket-fs PathLike object from a connection object
[This code](https://github.com/exasol/transformers-extension/blob/f31b7ea91db5e560b6207b6efcba52eddfd4c7ce/exasol_transformers_extension/utils/bucketfs_operations.py#L14) from the Transformers Extension can be reused.

### PR DESCRIPTION
**Pull request by ahsimb**
closes #74 
closes #75

### CODE REVIEW

File: exasol/python_extension_common/connections/bucketfs_location.py [hunk 1]
```diff
@@ -64,3 +40,175 @@ def create_bucketfs_location(**kwargs) -> bfs.path.PathLike:
...
+
+    db_type = _infer_backend(kwargs)
+    if db_type == _Backend.onprem:
+        return bfs.path.build_path(**_convert_onprem_bfs_params(kwargs))
+    else:
+        return bfs.path.build_path(**_convert_saas_bfs_params(kwargs))
+
+
+@dataclass
+class ConnObjectData:
```
Comment (Nicoretti) [inline exasol/python_extension_common/connections/bucketfs_location.py:112]:
```suggestion
class ConnectionInfo:
```

File: test/integration/connections/test_bucketfs_location.py [hunk 2]
```diff
@@ -1,86 +1,177 @@
...
         StdParams.bucket.name: 'default',
         StdParams.saas_url.name: 'my_saas_url',
     }
+
+
+@contextmanager
+def write_test_file(bfs_path: bfs.path.PathLike) -> None:
+    bfs_path.write(TEST_FILE_CONTENT)
+    try:
+        yield
```
Comment (Nicoretti) [inline test/integration/connections/test_bucketfs_location.py:32]:
```suggestion
    try:
        bfs_path.write(TEST_FILE_CONTENT)
        yield
```

File: test/integration/connections/test_bucketfs_location.py [hunk 3]
```diff
@@ -1,86 +1,177 @@
...
+    extra_params = {StdParams.path_in_bucket.name: 'test_create_location'}
+    bfs_path = create_bucketfs_location(**onprem_params, **extra_params)
+    with write_test_file(bfs_path):
+        validate_test_file(bfs_path)
+
+
+def test_create_bucketfs_location_saas_db_id(use_saas,
+                                             saas_params_with_id):
+    if not use_saas:
+        pytest.skip("The test is not configured to use SaaS.")
```
Comment (Nicoretti) [inline test/integration/connections/test_bucketfs_location.py]: Is it possible to achieve this decorator based?

**Review by Nicoretti — DISMISSED** [review]

File: test/integration/connections/test_bucketfs_location.py (hunk 3 above)
Comment (ahsimb) [reply test/integration/connections/test_bucketfs_location.py]: Possibly, but it's not trivial, as we need to look at the fixture.

**Review by Nicoretti — APPROVED** [review]

**Pull request merged**